### ISSUE TITLE
Why is it that when I train a pose model using my own dataset, all the training metrics appear normal, but during inference, not a single target or keypoint can be detected?

### ISSUE THREAD

**Issue by dadax7** [ultralytics/ultralytics#23996]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

<img width="2197" height="98" alt="Image" src="https://github.com/user-attachments/assets/d0836431-ce7c-4d51-b515-721c8b30f1e7" />

<img width="2223" height="90" alt="Image" src="https://github.com/user-attachments/assets/8167ba75-366f-44bd-9b25-8ca76ab6f7fe" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dadax7, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue moving quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a custom training ❓ question related to pose model behavior during inference, please provide as much information as possible, including dataset image examples, a few labeled samples, your dataset YAML, training logs, validation results, and the exact inference command or code used. If you believe this may be a 🐛 Bug Report, please also provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate efficiently 🔍.

Please also confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), and if possible include:
-Your Ultralytics package version
-Python and PyTorch versions
-OS and device details
-The exact model used for training and inference
-Whether the same images work during `val` but fail during `predict`
-A few representative train/val images and labels

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you post the training and inference code?

**Comment by dadax7** [comment]
@Y-T-G No，the code of train and test follows, and I can perform inference normally using the official pre-trained weights.
```from ultralytics import YOLO
model = YOLO("yolo26x-pose.yaml")
# Train the model
results = model.train(data="pose.yaml", epochs=100, imgsz=640, project='runs/pose', verbose=True, device="0", save_period=1, batch=32, pretrained=False)```
# test
model = YOLO(args.model)
results = model(
            str(img_path),
            imgsz=args.imgsz,
            conf=args.conf,
            save=False,  # 不使用自带保存
            verbose=False
        )

**Comment by Y-T-G** [comment]
How large is your dataset? Did you try running inference in images from your validation set?

**Comment by dadax7** [comment]
> How large is your dataset? Did you try running inference in images from your validation set?

about 4 million. Yes, I inference in images of coco_val datasets, and I print the torch output during inference, which find that values are all empty.

**Comment by glenn-jocher** [comment]
You’re training from scratch with `yolo26x-pose.yaml` and `pretrained=False`, so please first retry from `yolo26x-pose.pt`; also, if your dataset’s class/keypoint schema differs from COCO, empty results on `coco_val` are expected. Please run `model.val()` on `best.pt`, test on your own validation images with `conf=0.001`, and share the text output of `yolo checks`, your `pose.yaml`, and one image/label pair so we can verify the format against the [pose estimation docs](https://docs.ultralytics.com/tasks/pose/).

**Comment by dadax7** [comment]
> You’re training from scratch with `yolo26x-pose.yaml` and `pretrained=False`, so please first retry from `yolo26x-pose.pt`; also, if your dataset’s class/keypoint schema differs from COCO, empty results on `coco_val` are expected. Please run `model.val()` on `best.pt`, test on your own validation images with `conf=0.001`, and share the text output of `yolo checks`, your `pose.yaml`, and one image/label pair so we can verify the format against the [pose estimation docs](https://docs.ultralytics.com/tasks/pose/).

Thank you! I’d also like to ask, is it normal for the RLE_loss during validation to always be zero? But during the training process, it decreases normally and remains greater than zero.

**Comment by glenn-jocher** [comment]
Yes, that can be normal: in YOLO26 pose the `kpts_sigma` branch used for RLE is only returned in training mode, and `PoseLoss26.loss()` only fills `rle_loss` when `preds["kpts_sigma"]` exists, while validation is mainly evaluated with pose metrics like keypoint IoU and pose mAP, so `val/rle_loss=0` by itself does not indicate a bug; I’d treat it as separate from the empty-prediction issue and keep checking your dataset schema against the [pose loss reference](https://docs.ultralytics.com/reference/utils/loss/) and [pose validator reference](https://docs.ultralytics.com/reference/models/yolo/pose/val/). ([docs.ultralytics.com](https://docs.ultralytics.com/reference/nn/modules/head/))

**Comment by dadax7** [comment]
> Yes, that can be normal: in YOLO26 pose the `kpts_sigma` branch used for RLE is only returned in training mode, and `PoseLoss26.loss()` only fills `rle_loss` when `preds["kpts_sigma"]` exists, while validation is mainly evaluated with pose metrics like keypoint IoU and pose mAP, so `val/rle_loss=0` by itself does not indicate a bug; I’d treat it as separate from the empty-prediction issue and keep checking your dataset schema against the [pose loss reference](https://docs.ultralytics.com/reference/utils/loss/) and [pose validator reference](https://docs.ultralytics.com/reference/models/yolo/pose/val/). ([docs.ultralytics.com](https://docs.ultralytics.com/reference/nn/modules/head/))

Thank you very much! I've noticed that in the validation set, some images can be inferred normally, while others produce no output at all. What are some improvement methods for this kind of problem?

**Comment by dadax7** [comment]
> You’re training from scratch with `yolo26x-pose.yaml` and `pretrained=False`, so please first retry from `yolo26x-pose.pt`; also, if your dataset’s class/keypoint schema differs from COCO, empty results on `coco_val` are expected. Please run `model.val()` on `best.pt`, test on your own validation images with `conf=0.001`, and share the text output of `yolo checks`, your `pose.yaml`, and one image/label pair so we can verify the format against the [pose estimation docs](https://docs.ultralytics.com/tasks/pose/).

this is my pose.yaml,  is it correct? 
```# Keypoints
kpt_shape: [17, 3] # number of keypoints, number of dims (2 for x,y or 3 for x,y,visible)
flip_idx: [0, 2, 1, 4, 3, 6, 5, 8, 7, 10, 9, 12, 11, 14, 13, 16, 15]

# Classes
# nc: 4
# names: ['person', 'head', 'face', 'upper_body']
names:
  0: person
  1: head
  2: face
  3: upper_body

# Keypoint names per class
kpt_names:
  0:
    - nose
    - left_eye
    - right_eye
    - left_ear
    - right_ear
    - left_shoulder
    - right_shoulder
    - left_elbow
    - right_elbow
    - left_wrist
    - right_wrist
    - left_hip
    - right_hip
    - left_knee
    - right_knee
    - left_ankle
    - right_ankle ```

**Comment by glenn-jocher** [comment]
Thanks — `kpt_shape` and `flip_idx` look fine, but if this is the full file the class section looks inconsistent: you define 4 classes in `names` and only provide pose keypoints for class `0`. If your labels are human pose only, keep it single-class (`0: person`); otherwise make sure every labeled class follows the same pose schema and every `.txt` row has exactly `17*3` keypoint values with visibility `0/1/2`, then delete any `labels/*.cache` files and retrain from `yolo26x-pose.pt`. For the images that return nothing, first retry those same validation images with `conf=0.001`; if they still fail, please share one missed image and its label so we can compare it against the [pose docs](https://docs.ultralytics.com/tasks/pose/) and the official [coco-pose.yaml example](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/coco-pose.yaml). ([github.com](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/coco-pose.yaml))
